gremlin-scala/src/test/scala/gremlin/scala/TraversalSpec.scala

```diff
@@ -407,6 +408,56 @@ class TraversalSpec extends WordSpec with Matchers {
     }
   }
 
+  "coalesce" should {
+    "evaluate traversals and return first with value" in new Fixture {
+
+      def path2String(path: Path) = path.objects().map(_.toString)
+
+      val traversal1 = graph.V(1)
+        .coalesce(
+          _.outE("knows"),
+          _.outE("created")
+        )
+        .inV()
+        .path().by("name").by(T.label)
+
+      traversal1.toList().map(path2String) shouldBe
+        Seq(Seq("marko", "knows", "vadas"), Seq("marko", "knows", "josh"))
+
+
+      val traversal2 = graph.V(1)
+        .coalesce(
+          _.outE("created"),
+          _.outE("knows")
+        )
+        .inV()
+        .path().by("name").by(T.label)
+
+      traversal2.toList().map(path2String) shouldBe Seq(Seq("marko", "created", "lop"))
+
+      graph.graph.V(1).property(Nickname, "okram").iterate()
+
+      val traversal3 = graph.V
+        .hasLabel("person")
+        .coalesce(
+          _.value(Nickname),
+          _.value(Name)
+        )
+
+      traversal3.toSet() shouldBe Set("okram", "vadas", "josh", "peter")
```

Comment:
i think it would make sense to split this up into three separate tests which are easier to read.